### ISSUE TITLE
docker engine stopped

### ISSUE THREAD

**Issue by ARRed98** [docker/for-win#14210]
### Description

For the past week, my Docker has been down. I've tried purging all the data, reinstalling Docker, and reinstalling WSL, but nothing works. I'm on Windows 11, with the latest WSL and the latest version of Docker. When I switch to Windows containers, it works. However, in Linux containers, the engine just doesn't run or gets stuck on "engine starting" after a data purge. I've pretty much run out of ideas and tried everything I knew.

When I disable wsl2 and run through hyper v the docker engine starts right away

### Reproduce

Install docker
Launch docker
Docker engine doesn't start

### docker version

```bash
Client:
 Version:           27.0.3
 API version:       1.46
 Go version:        go1.21.11
 Git commit:        7d4bcd8
 Built:             Sat Jun 29 00:03:32 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

command get stuck and doesn't print anything unless i cancel. So it doesn't output the information as it should
```

### docker info

```bash
doesn't output info
```

### Diagnostics ID

A9788111-82F3-4995-AB8E-7C8332706713/20240722073457

**Comment by ARRed98** [comment]
downgrading from the pre-release to latest version of wsl fixed the problem for me

**Comment by Sachi9892** [comment]
Try this = Delete the .docker file and then restart docker desktop . you can find the .docker file mostly at C:/Yourname/ .

**Comment by ben-joostens** [comment]
> Try this = Delete the .docker file and then restart docker desktop . you can find the .docker file mostly at C:/Yourname/ .

`c:\Username` does not exist, do you mean `c:\Users\Username`? Also, in that location there is only a `.docker` directory, not a file, is deleting that what fixed it for you?

**Comment by Sachi9892** [comment]
Yes in c:\Users\Username you will find  .docker directory . I deleted .docker directory and now i'm able to run my docker desktop.. it worked for me

**Comment by lorenrh** [comment]
this seems related to https://github.com/docker/for-win/issues/14207 🤔

**Comment by dev-gilbert** [comment]
<img width="960" alt="Capture" src="https://github.com/user-attachments/assets/bf2431a5-91b8-48db-9f1c-a2f04a00ed9a">
 I deleted .docker the problem was not fixed AND IT IS TELLING ME TO ENABLE VTX in BIOS and it was already enabled

**Comment by elsaco** [comment]
Fixed in `wsl-2.3.13` release!

**Comment by Farzane2630** [comment]
> Try this = Delete the .docker file and then restart docker desktop . you can find the .docker file mostly at C:/Yourname/ .

@Sachi9892 this worked for me. thank you

**Comment by dev-gilbert** [comment]
Thank you

**Comment by gicharukabare** [comment]
Just installed Docker , but the Engine stops at every attempt to load.

C:\Windows\System32>docker version
Client:
 Version:           27.4.0
 API version:       1.47
 Go version:        go1.22.10
 Git commit:        bde2b89
 Built:             Sat Dec  7 10:40:21 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

**Comment by Tiansang-Psy** [comment]
> Yes in c:\Users\Username you will find .docker directory . I deleted .docker directory and now i'm able to run my docker desktop.. it worked for me

Great! this worked for me. thank you!!!!

**Comment by ubaydimse** [comment]
1. make sure virtualization active in bios
2. enable virtualization in optional feature
3. cmd -> ipconfig /flushdns
    cmd -> netsh winsock reset
4. delete directory in C:\Users\<username>\ .docker
5. cmd -> wsl --list --verbose
make sure you only see docker-desktop with state stopped and version 2
6. cmd -> wsl --update
    cmd -> wsl --shutdown
7. restart your pc and run docker desktop

that's it

**Comment by iSplasher** [comment]
Had this issue after updating to "Docker Desktop 4.39.0" with "Docker version 28.0.1, build 068a01e", and I had just updated wsl earlier. None of the proposed solutions here worked for me.
What ended up working was remembering that I had enabled WSL Integration for a distro in Docker Desktop settings, so I went ahead and ran `sudo apt update && sudo apt upgrade` in wsl and ensured running `docker --version` inside wsl reported the same version and build as running the command outside wsl.
The issue went away after that.

**Comment by patricknelson** [comment]
Not sure why yet, but I also had the same problem. It resolved itself when I just manually logged into the `docker-desktop` distro in WSL2, i.e.

```
wsl -d docker-desktop
```

After that, Docker Desktop unfroze, reported an error, then restarted fine. Prior to that it was just halted on a "Docker engine stopped" status with no end in sight. Also Docker Desktop 4.39.0 (184744) and same CLI version (Docker version 28.0.1, build 068a01e). Cannot upgrade yet as I don't have local admin rights on this particular machine.

**Comment by electroix** [comment]
> Yes in c:\Users\Username you will find .docker directory . I deleted .docker directory and now i'm able to run my docker desktop.. it worked for me

It works for me too. Thank you!

**Comment by PikarryPham** [comment]
> Yes in c:\Users\Username you will find .docker directory . I deleted .docker directory and now i'm able to run my docker desktop.. it worked for me

This also works for me, although I don't know the reason. Thank you so much.

**Comment by huy19th** [comment]
> Yes in c:\Users\Username you will find .docker directory . I deleted .docker directory and now i'm able to run my docker desktop.. it worked for me

This also works for me, thanks!

**Comment by boyrabbit233** [comment]
Install the latest version of wsl worked for me. https://github.com/microsoft/WSL/releases/download/2.6.0/wsl.2.6.0.0.x64.msi
